controllers/harbor/provision.go

```diff
@@ -0,0 +1,241 @@
+package harbor
+
+import (
+	goharborv1 "github.com/goharbor/harbor-cluster-operator/api/v1"
+	"github.com/goharbor/harbor-cluster-operator/controllers"
+	"github.com/goharbor/harbor-cluster-operator/controllers/image"
+	"github.com/goharbor/harbor-cluster-operator/lcm"
+	"github.com/goharbor/harbor-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Provision will create a new Harbor CR.
+func (harbor *HarborReconciler) Provision() (*lcm.CRStatus, error) {
+	harborCR := harbor.newHarborCR()
+	err := harbor.Create(harborCR)
+	if err != nil {
+		return harborCRNotReadyStatus("", ""), err
+	}
+	return harborCRUnknownStatus(), err
+}
+
+// newHarborCR will create a new Harbor CR controlled by harbor-operator
+func (harbor *HarborReconciler) newHarborCR() *v1alpha1.Harbor {
+	namespacedName := harbor.getHarborCRNamespacedName()
+
+	return &v1alpha1.Harbor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      namespacedName.Name,
+			Namespace: namespacedName.Namespace,
+			Labels: map[string]string{
+				controllers.HarborClusterNameLabel: harbor.HarborCluster.Name,
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(harbor.HarborCluster, goharborv1.HarborClusterGVK),
```

Comment:
Maybe we can use SetControllerReference set OwnerReference.

Reply:
`SetControllerReference` will return an error when there is another OwnerReference with Controller flag set. But we don't need to prevent this error. So `NewControllerRef` will be simpler.